doc/user/content/sql/create-source/postgres.md

```diff
@@ -36,132 +36,107 @@ Field | Value type | Description
 
-4. Create a publication containing all the tables you wish to query in Materialize:
+```sql
+CREATE VIEWS FROM SOURCE mz_source
+(public.foo AS foo, otherschema.foo AS foo2);
+```
 
-    *For all tables in Postgres:*
-    ```sql
-    CREATE PUBLICATION mz_source FOR ALL TABLES;
-    ```
+#### Creating materialized views
 
-    *For specific tables:*
-    ```sql
-    CREATE PUBLICATION mz_source FOR TABLE table1, table2;
-    ```
+To produce correct results, Postgres sources can only be materialized _once_. As soon as you define a materialized view, Materialize:
 
-### Creating a source
+1. Creates a replication slot in the upstream Postgres database (see [Postgres replication slots](#postgres-replication-slots)). The name of the replication slots created by Materialize is prefixed with `materialize_` for easy identification.
 
-Once you have set up Postgres, you can create the source in Materialize.
+1. Performs an initial, snapshot-based sync of the tables in the publication before it starts ingesting change events.
 
-```sql
-CREATE MATERIALIZED SOURCE "mz_source" FROM POSTGRES
-CONNECTION 'host=postgres port=5432 user=host sslmode=require dbname=postgres'
-PUBLICATION 'mz_source';
-```
+   **Note:** During this phase, **disk space** consumption may increase before returning to a steady state. To profile disk usage, see [Troubleshooting](/ops/troubleshooting/#how-much-disk-space-is-materialize-using).
 
-This creates a source that...
+1. Incrementally updates the view as new change events stream in as a result of `INSERT`, `UPDATE` and `DELETE` operations in the upstream Postgres database.
 
-- Connects to a Postgres server
-- Contains raw data from all of the tables that went into the publication
-- Needs to broken out into more usable views that reproduce the original Postgres tables
+##### Postgres replication slots
 
-### Creating views
+Each Materialize replication slot can be used to source data for a single materialized view. You can create multiple non-materialized views for the same replication slot using the [`CREATE VIEWS`](/sql/create-views) statement.
 
-Once you have created the Postgres source, you need to create views that represent the upstream publication's original tables.
+{{< warning >}}
+Make sure to delete any replication slots if you stop using Materialize or if either your Materialize or Postgres instances crash.
+{{< /warning >}}
 
-*Create views for all tables included in the Postgres publication*
+If you stop Materialize or delete the materialized view without dropping the Postgres source first, the upstream replication slot isn't deleted and will continue to accumulate data. In such cases, you should manually delete the replication slot to recover memory and avoid degraded performance in the upstream database.
```

Comment:
```suggestion
If you stop Materialize or delete the materialized view without also dropping the Postgres source, the upstream replication slot will linger and will continue to accumulate data so that the source can resume in the future. To avoid unbounded disk space usage make sure to use `DROP SOURCE` or manually delete the replication slot in case you have deleted the Materialize instance.
```

We should also mention the `max_slot_wal_keep_size` parameter https://www.postgresql.org/docs/13/runtime-config-replication.html#GUC-MAX-SLOT-WAL-KEEP-SIZE

Reply:
Ah, great reminder! I'll just adapt the snippet from [this discussion](https://github.com/MaterializeInc/materialize/discussions/7281) and blend it in:

> **For scenarios where the processing of the replication stream lags in Materialize, is there a way to guarantee the stability of the upstream Postgres database?**
> This is a known pain point in Postgres, and Materialize itself doesn't provide any mechanism to prevent it. From Postgres v13+, though, you can set a reasonable value for [max_slot_wal_keep_size](https://www.postgresql.org/docs/devel/runtime-config-replication.html#GUC-MAX-SLOT-WAL-KEEP-SIZE) to limit the amount of storage used by replication slots; this is recommended for production setups. If the replication slot happens to fall behind and hit the defined limit, the upstream database will just start recycling WAL files and Materialize will stop replicating.